data_safe_haven/commands/pulumi.py

```diff
@@ -0,0 +1,80 @@
+"""Interact with the Pulumi CLI using DSH projects"""
+
+from enum import UNIQUE, StrEnum, auto, verify
+from typing import Annotated
+
+import typer
+from rich import print
+
+from data_safe_haven.config import Config, DSHPulumiConfig
+from data_safe_haven.context import ContextSettings
+from data_safe_haven.external import GraphApi
+from data_safe_haven.functions import sanitise_sre_name
+from data_safe_haven.infrastructure import SHMProjectManager, SREProjectManager
+
+pulumi_command_group = typer.Typer()
+
+
+@verify(UNIQUE)
+class ProjectType(StrEnum):
+    SHM = auto()
+    SRE = auto()
+
+
+@pulumi_command_group.command()
+def run(
+    project_type: Annotated[
+        ProjectType,
+        typer.Argument(
+            # metavar="{SHM|SRE}",
```

Comment:
Do we need this comment?

Reply:
No, I was trying to make the help text clearer, but it didn't work as I expected it to. Possibly a conflict with the rich rendering. In any case it isn't needed now.